backend-rust/Dockerfile

```diff
@@ -0,0 +1,23 @@
+ARG build_image=rust:1.82-bookworm
+ARG base_image=debian:bookworm-slim
+FROM ${build_image} AS build
+
+WORKDIR /usr/app
+
+COPY Cargo.toml Cargo.lock ./
+COPY src ./src
+COPY concordium-rust-sdk /usr/app/concordium-rust-sdk
+COPY .sqlx .sqlx
+RUN cargo install sqlx-cli --no-default-features --features "postgres"
+RUN cargo build --release --locked
+
+
+FROM ${base_image}
+
+WORKDIR /usr/app
+
+COPY migrations /usr/app/migrations
+COPY --from=build /usr/app/target/release/ccdscan-api /usr/app/target/release/ccdscan-indexer /usr/bin/
+COPY --from=build /usr/local/cargo/bin/sqlx /usr/local/bin/sqlx
+
+RUN chmod +x /usr/bin/ccdscan-api /usr/bin/ccdscan-indexer
```

Comment:
I  think there should be two releases/dockerfiles (`indexer` and `graph-api` service). This means two independent images in the end since these services are run independently (e.g. we want to run several `graph-api` instances but only a single `indexer`).

similar to:
https://github.com/Concordium/concordium-dapp-examples/tree/main/trackAndTrace/dockerfiles

Reply:
I can indeed do both based on what I have here.

There is a few issues with the above mentioned dockerfiles though:

1. We are building the same code twice which is inefficient
2. We are building the same code without locking. This means that you have no guarantee that the code is actually the same.

Reply:
https://github.com/Concordium/concordium-scan/actions/runs/12048317579

Can see the effect of the altercation in the above mentioned link.

I just ended up using caching.